src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -201,6 +201,15 @@ public ReadPileup getPileupForSample(final String sample, final SAMFileHeader he
     }
 
     /**
+     * Iterator over sorted by read start PileupElements.
+     */
+    public Iterator<PileupElement> sortedIterator() {
+        return getElementStream()
+                .sorted((l, r) -> Integer.compare(l.getRead().getStart(), r.getRead().getStart()))
+                .iterator();
+    }
```

Comment:
Looks good now -- if you can just add a direct unit test in `ReadPileupUnitTest` for the new `sortedIterator()` method, then we can merge this.